test/dualnamedarray_unit.C

```diff
@@ -45,10 +47,10 @@ namespace MetaPhysicL {
 } // namespace MetaPhysicL
 #endif
 
+#include "metaphysicl/sparsenumbervector.h"
 #include "metaphysicl/dualnamedarray.h"
 #include "metaphysicl/metaphysicl_asserts.h"
 #include "metaphysicl/metaphysicl_exceptions.h"
-#include "metaphysicl/sparsenumbervector.h"
```

Comment:
Yes, annoyingly!  IMHO the *right* thing to do would be to break out a new `sparsenumbervector_decl.h`, but for the existing use cases this was sufficient.  When you have a `Foo<Bar> foobar;` and the `Foo` code calls functions with `Bar` overloads, you *must* have the overload declarations included first, or they don't get seen.

I reordered these last year, having forgotten about that dependency, but I got away with it because we haven't run with VexCL (which those two unit test files need) in more than a year.

Reply:
> Yes

Or I guess to be more accurate, if the answer is No then I don't know what the alternative is.  In the most annoying case, I've only ever gotten `Foo<Bar<Foo<Bar>>>` code to compiler properly by breaking headers into e.g. a `foo_decl.h` of just declarations (so `Bar` definitions can be included after that and will see that `Foo` overloads exist) along with a `foo.h` of definitions (so `Foo` code can be included after `bar_decl.h` and see `Bar` overloads).